### PR TITLE
Flow run logs csv download

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Adds a new endpoint `/api/flow_runs/{id}/download-logs-csv` which downloads a csv of all logs a flow run created. This csv looks like
```
timestamp,level,flow_run_id,task_run_id,message
2024-07-22 17:58:14.037664+00:00,20,32ded098-2633-414e-90cf-242af9d6d2a8,,"Lorem ipsum http://google.com hello world https://example.com cache_key_fn, type: <class 'function'>, fn: <function task_input_hash at 0x7f165f8fba30>"
2024-07-22 17:58:15.039871+00:00,20,32ded098-2633-414e-90cf-242af9d6d2a8,,"Lorem ipsum http://google.com hello world https://example.com cache_key_fn, type: <class 'function'>, fn: <function task_input_hash at 0x7f165f8fba30>"
2024-07-22 17:58:16.080519+00:00,20,32ded098-2633-414e-90cf-242af9d6d2a8,,Finished in state Completed()

```

### Example
Going to `"api/flow_runs/{id}/download-logs-csv` downloads a csv file to the browser.

### Checklist

- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
- [x] This pull request references any related issue by including "closes `<link to issue>`"
  
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

**Review by cicdw — CHANGES_REQUESTED** [review]
Totally understand the motivation here, and your implementation makes a lot of sense if it ran client-side.  However, `while True:` style loops like this have big performance implication (and risk) when run server side, and in addition this could eat up a lot of memory depending on the size of the logs being requested, so I think we're going to need to consider other alternatives.

@chrisguidry or @zangell44 let me know if you disagree with my take here

**Comment by chrisguidry** [comment]
> Totally understand the motivation here, and your implementation makes a lot of sense if it ran client-side. However, `while True:` style loops like this have big performance implication (and risk) when run server side, and in addition this could eat up a lot of memory depending on the size of the logs being requested, so I think we're going to need to consider other alternatives.
> 
> @chrisguidry or @zangell44 let me know if you disagree with my take here

It looks like this is using a `StreamingResponse` with a generator, so it _should_ be okay from a memory perspective since it should only have a hard references to that 1000 logs at a time.

**Review by chrisguidry — COMMENTED** [review]
I _think_ we'll be good since you're using the `StreamingResponse`, but can we get a unit test in for this one?

**Comment by cicdw** [comment]
Awesome, makes sense - deferring my review to @chrisguidry in that case 🫡

**Comment by pleek91** [comment]
@cicdw and @chrisguidry thank you! I'll work on some tests. Also @chrisguidry and @zangell44 curious about cloud and if this same solution would work there.

**Comment by chrisguidry** [comment]
Yep I think streaming would work just fine on Prefect Cloud too, although there are a few more layers of load balancers/proxies involved.  When we get there, let's set aside some time to design a load test for it

File: tests/server/orchestration/api/test_flow_runs.py [hunk 1]
```diff
@@ -2693,3 +2694,99 @@ async def test_read_subflow_runs_non_existant(
...
+        monkeypatch.setattr(
+            "prefect.server.api.flow_runs.FLOW_RUN_LOGS_CSV_PAGE_LIMIT", 3
+        )
+
+        async with client.stream(
+            "GET", f"/flow_runs/{flow_run_1.id}/download-logs-csv"
+        ) as response:
+            response_body = [
+                str(chunk, "UTF-8") async for chunk in response.aiter_bytes()
+            ]
```
Comment (chrisguidry) [inline tests/server/orchestration/api/test_flow_runs.py:2782]: There we go, nice!

**Review by chrisguidry — APPROVED** [review]
Looks great, thanks for that extra time on the testing!

**Review by cicdw — APPROVED** [review]

**Review by discdiver — APPROVED** [review]

**Pull request merged**